### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by elbing**
- wstat is used now from libc directly, no wrapper needed.
- Duplicated rename.c to single one in ap/plan9/rename.c,
  stdio version was using link/unlink and link is a dummy here.
- Fixed rename(), which has a check through _dirwstat that wasn't
  working.
- Fixed a value in limits.h which was for 32 bit and was colliding
  with an internal header.
- cpp wrapper for encapsulate funcs calling common syscalls that could
  collide with ANSI/POSIX implementations are not now in uppercase, using
  underscores instead.
- Properly remove() sources files renaming.
- Added float flag for clang which was breaking binaries (never happened before).

Change-Id: I1209c54d7c3fa856a2039ef7833189bd568dfcab

### CODE REVIEW

**Comment by elbing** [comment]
Sadly, this doesn't solve "too many opened files" error.

File: lib/ap/plan9/rename.c [hunk 1]
```diff
@@ -43,15 +43,15 @@ rename(const char *from, const char *to)
 	f = f? f+1 : (char *)from;
 	t = t? t+1 : (char *)to;
 	n = 0;
-	if(f-from==t-to && strncmp(from, to, f-from)==0){
-		/* from and to are in same directory (we miss some cases) */
+	/* if(f-from==t-to && strncmp(from, to, f-from)==0){
```
Comment (rminnich) [inline lib/ap/plan9/rename.c]: nasty. Hard to follow. Would be nicer if you just did this:
if (0 && (f-from == etc. etc.)
Then the code would compile, always, and we could come back and fix it later.

Comment (elbing) [inline lib/ap/plan9/rename.c]: I have nightmares with that since gcc-6, but ok. Style!

**Comment by elbing** [comment]
Suggestion done.

File: lib/ap/plan9/rename.c [hunk 2]
```diff
@@ -43,7 +43,7 @@ rename(const char *from, const char *to)
 	f = f? f+1 : (char *)from;
 	t = t? t+1 : (char *)to;
 	n = 0;
-	if(f-from==t-to && strncmp(from, to, f-from)==0){
+	if(0 && (f-from==t-to && strncmp(from, to, f-from)==0)){
```
Comment (rminnich) [inline lib/ap/plan9/rename.c]: I'm going to LGTM but you might want to put a comment here explaining why you put 0 && to disable this path.

**Comment by rminnich** [comment]
LGTM

File: lib/ap/plan9/rename.c (hunk 2 above)
Comment (elbing) [inline lib/ap/plan9/rename.c]: I said that: "Fixed rename(), which has a check through _dirwstat that wasn't working." Reasons about why is not working seems being related with permissions for APEX over the FS, what is causing that _dirwstat always was returning -1. But this is a theory, this should be tested with a properly FS with auth enabled.

Comment (elbing) [inline lib/ap/plan9/rename.c]: Ah, I understand, on the code. Ok. Well, next round.
